### PR TITLE
Fix cannot clear value when using free text mode

### PR DESCRIPTION
**Pull request by Sakchai-Refinitiv**
## Description
The value cannot be cleared by using programmatically when enabled free text mode and set custom text to the input.

Fixes https://jira.refinitiv.com/browse/ELF-1336

## Type of change
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist
- [x]  My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by goremikins** [comment]
It should be possible to not just clear the free text value, but to set it as well, if that value does not exist in the set. @TremayneChrist , do you agree?

**Comment by TremayneChrist** [comment]
Yes. If free text mode is set, you should be able to do...

```js
el.value = '';
el.value = 'any string value I like';
```

**Comment by goremikins** [comment]
For instance like this:
```
     else {
       // Clone value arrays
       const newValues = values.slice(0, this.multiple ? values.length : 1);
      const oldValues = this.values.slice();
       // Create comparison strings to check for differences
       const newComparison = newValues.sort().toString();
       const oldComparison = oldValues.sort().toString();
       // Should we update the selection state?
       if (newComparison !== oldComparison) {
         this.updateComposerValues(newValues);
        if (this.freeText) {
          this.freeTextValue = !this.composerValues.length ? newComparison : '';
       }

         this.requestUpdate('values', oldValues);
       }

```

**Comment by Sakchai-Refinitiv** [comment]
Hi All, thank you for your review, I have already patched it.

**Comment by Theeraphat-Sorasetsakul** [comment]
LGTM now. Could you please double check @goremikins @TremayneChrist ?

**Review by Theeraphat-Sorasetsakul — APPROVED** [review]

File: packages/elements/src/combo-box/index.ts [hunk 1]
```diff
@@ -279,13 +279,19 @@ export class ComboBox<T extends DataItem = ItemData> extends FormFieldElement {
...
       const newComparison = newValues.sort().toString();
       const oldComparison = oldValues.sort().toString();
-      // Should we update the selection state?
+
+      // Update the selection state when found new value
       if (newComparison !== oldComparison) {
         this.updateComposerValues(newValues);
+
+        if (this.freeText) {
+          this.freeTextValue = !this.composerValues.length ? newComparison : '';
```
Comment (goremikins) [inline packages/elements/src/combo-box/index.ts:292]: I think here is more correct to set newValues[0]: `this.freeTextValue = !this.composerValues.length ? newValues[0] : '';`

Comment (Sakchai-Refinitiv) [reply packages/elements/src/combo-box/index.ts]: Resolved

File: packages/elements/src/combo-box/index.ts [hunk 2]
```diff
@@ -279,13 +279,19 @@ export class ComboBox<T extends DataItem = ItemData> extends FormFieldElement {
...
       const newComparison = newValues.sort().toString();
       const oldComparison = oldValues.sort().toString();
-      // Should we update the selection state?
+
+      // Update the selection state when found new value
       if (newComparison !== oldComparison) {
         this.updateComposerValues(newValues);
+
+        if (this.freeText) {
+          this.freeTextValue = !this.composerValues.length ? newValues[0] : '';
```
Comment (TremayneChrist) [inline packages/elements/src/combo-box/index.ts:292]:
```suggestion
          // free text mode is only supported in single selection mode
          // so if there is no valid selection in the composer, we can assume
          // the first item can be used as the free text item.
          this.freeTextValue = !this.composerValues.length ? newValues[0] : '';
```

**Comment by Theeraphat-Sorasetsakul** [comment]
Any update? Can this pull request be merged? @Sakchai-Refinitiv @TremayneChrist

**Comment by Sakchai-Refinitiv** [comment]
> Any update? Can this pull request be merged? @Sakchai-Refinitiv @TremayneChrist

I am waiting for @TremayneChrist to confirm for merge.

**Review by TremayneChrist — APPROVED** [review]

**Pull request merged**